src/dune_sexp/versioned_file.ml

```diff
@@ -61,17 +61,20 @@ struct
         match Atom.parse ver with
         | Some atom -> atom
         | None ->
-          let hints =
-            match Table.find langs name with
-            | None -> []
-            | Some lang ->
-              let latest = Syntax.greatest_supported_version_exn lang.syntax in
-              [ Pp.textf "lang dune %s" (Syntax.Version.to_string latest) ]
-          in
-          User_error.raise
-            ~loc:ver_loc
-            ~hints
-            [ Pp.text "Invalid version. Version must be two numbers separated by a dot." ]
+          (* Atom.parse failed - likely due to non-ASCII characters *)
+          let has_non_ascii = String.exists ver ~f:(fun c -> Char.code c >= 128) in
+          if has_non_ascii
+          then
+            User_error.raise
+              ~loc:ver_loc
+              [ Pp.text
+                  "Invalid atom: contains non-ASCII character(s). Atoms must only \
+                   contain ASCII characters."
+              ]
+          else
+            Code_error.raise
+              "Atom.parse failed for unexpected reason"
```

Comment:
Is it really unexpected? Can't it happen for some other invalid character? A regular error should suffice here.